R/translate.R

```diff
@@ -1,7 +1,3 @@
-obj_maybe_translate_encoding <- function(x) {
-  .Call(vctrs_maybe_translate_encoding, x)
-}
-
-obj_maybe_translate_encoding2 <- function(x, y) {
-  .Call(vctrs_maybe_translate_encoding2, x, y)
+proxy_normalize_encoding <- function(x) {
```

Comment:
The `vec_proxy_` prefix would be more consistent?

Reply:
Hmm, would it? This function has to be called on a proxy. `vec_proxy_equal()`, for example, is called on a vector but returns a proxy.

Reply:
I should make the input argument here at the R level `proxy` rather than `x`, like I did at the C level

Reply:
> Hmm, would it?

I think so. A proxy is still a vector. `vec_proxy_assign()` takes a proxy. `vec_proxy_typeof()` returns the type of a proxy.

Reply:
Hmm, on second thought I'm no longer convinced that this _has_ to be called on a proxy anymore. It previously had to be because of how it handled data frames, but now we treat those like lists. It should really be able to be called on any vector without issue.

I think that I can call this `vec_normalize_encoding(x)`, with a note that it is generally called on a proxy.

That said, even though this function would "work" on any vector, sometimes we have to call it on the proxy to get "correct" behavior. Consider a list-rcrd with a vectorized character attribute. `vec_normalize_encoding()` doesn't translate attributes, so you'd have to call `vec_proxy_equal()` before `vec_normalize_encoding()` to promote the attribute to a data frame column first, so it gets translated. I think I can just add a note about this.